Description: Fixed wrong aspect ratio in transcoded image from webcam.
Origin: upstream, https://trac.videolan.org/vlc/ticket/4312
Bug-Ubuntu: https://launchpad.net/bugs/672304
diff -Nru vlc-1.1.4.orig//modules/stream_out/transcode/video.c vlc-1.1.4/modules/stream_out/transcode/video.c
--- vlc-1.1.4.orig//modules/stream_out/transcode/video.c	2010-07-24 06:17:44.000000000 -0700
+++ vlc-1.1.4/modules/stream_out/transcode/video.c	2010-11-17 14:25:10.115084004 -0800
@@ -435,6 +435,11 @@
     if( id->p_encoder->fmt_out.video.i_sar_num <= 0 ||
         id->p_encoder->fmt_out.video.i_sar_den <= 0 )
     {
+	vlc_ureduce( &id->p_decoder->fmt_out.video.i_sar_num, 
+ 	                &id->p_decoder->fmt_out.video.i_sar_den, 
+ 	                id->p_decoder->fmt_out.video.i_sar_num, 
+ 	                id->p_decoder->fmt_out.video.i_sar_den, 
+ 	                0 ); 
         id->p_encoder->fmt_out.video.i_sar_num = id->p_decoder->fmt_out.video.i_sar_num * i_src_width / i_dst_width;
         id->p_encoder->fmt_out.video.i_sar_den = id->p_decoder->fmt_out.video.i_sar_den * i_src_height / i_dst_height;
     }
